staging/src/k8s.io/legacy-cloud-providers/gce/gce_zones.go

```diff
@@ -79,7 +80,7 @@ func (g *Cloud) ListZonesInRegion(region string) ([]*compute.Zone, error) {
 	defer cancel()
 
 	mc := newZonesMetricContext("list", region)
-	list, err := g.c.Zones().List(ctx, filter.Regexp("region", g.getRegionLink(region)))
+	list, err := g.c.Zones().List(ctx, filter.Regexp("region", fmt.Sprintf(".*/regions/%s", region)))
```

Comment:
why not fix the function `getRegionLink` instead. Also a comment is worth.

Reply:
we cannot use regexp in getRegionLink, which is also used by other functions.